### PR TITLE
GTFS: Reading and Writing 

### PR DESCRIPTION
**Pull request by chris-jpark**
# Issue

Cleaning up a [ past PR ](https://github.com/valhalla/valhalla/pull/3660) that included past commits. 
Tests the ability to read and write GTFS files using the [just_gtfs](https://github.com/mapsme/just_gtfs/tree/cpp14) submodule.

### CODE REVIEW

File: test/data/gtfs_test/frequencies.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+trip_id,start_time,end_time,headway_secs,exact_times
+bar,00:00:00,02:00:00,1800,0
```
Comment (kevinkreiser) [inline test/data/gtfs_test/frequencies.txt:2]: should also probably have an example where its schedule based (ie exact_times == 1)

File: test/data/gtfs_test/calendar_dates.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+service_id,date,exception_type
+bon,20220202,1
```
Comment (kevinkreiser) [inline test/data/gtfs_test/calendar_dates.txt:2]: need one of type "added" and of type "removed"

File: test/gurka/test_gtfs.cc [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+  Feed feed;
+
+  // write agency.txt
+  struct Agency ttc {
+    .agency_id = "TTC", .agency_name = "Toronto Commission", .agency_url = "http://www.ttc.ca",
+    .agency_timezone = "America/Toronto"
+  };
+  feed.add_agency(ttc);
+  feed.write_agencies(("../build/test/data/gtfs_test/"));
```
Comment (kevinkreiser) [inline test/gurka/test_gtfs.cc:28]: these are run from inside wherever your build folder is
```suggestion
  feed.write_agencies(("test/data/gtfs_test/"));
```

File: test/gurka/test_gtfs.cc [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+using namespace gtfs;
+using namespace valhalla;
+
+#define tripOneID "bar"
+#define tripTwoID "barbar"
+#define stopOneID "foo"
+#define stopTwoID "foo2"
+#define shapeOneID "square"
+#define serviceOneID "bon"
```
Comment (kevinkreiser) [inline test/gurka/test_gtfs.cc:15]: we shouldnt use macros for constants these should look like:

`const std::string tripOneID = "bar";`

File: src/mjolnir/valhalla_fetch_transit.cc [hunk 5]
```diff
@@ -25,18 +25,33 @@
...
+// to include stop_ids or stops
+typedef struct tileTransitInfo {
+  GraphId graphId;
+  Stops tile_stops;
+  Trips tile_trips;
+  Routes tile_routes;
+  std::vector<Id> service_ids;
+  Agencies tile_agencies;
+  int size;
+} tileTransitInfo;
```
Comment (nilsnolde) [inline src/mjolnir/valhalla_fetch_transit.cc:54]: let's focus on one thing per PR. this PR is only about proving we can read/write with `just_gtfs`. we did that in the tests. after we're happy with that, we'll merge and start the work on "fetching" tiles.

**Comment by nilsnolde** [comment]
and can you make sure your new test file is running properly locally? it failed previously with smth like "mismatched arguments" for one of the gtfs functions. it failed CI.

**Comment by chris-jpark** [comment]
> 

this is strange, since the tests ran smoothly locally. I'll try to make the code create the directory "gtfs_test"and place it inside the data folder, it seemed like the code didn't know how to write the files until I created the directory manually.

**Comment by chris-jpark** [comment]
Nevermind, it failed CI because I edited the 'valhalla_fetch_transit' forgetting to checkout to another branch.

**Comment by nilsnolde** [comment]
> it failed CI because I edited the 'valhalla_fetch_transit'

jep, that's the recent failure and will be gone after cleaning up. I mean this one: https://app.circleci.com/pipelines/github/valhalla/valhalla/9229/workflows/9c637106-624f-4be2-a948-a621efcf7205/jobs/53269

> the tests ran smoothly locally

it doesn't for me and fails with the same error as in the CI I referenced:
```
[build] /home/nilsnolde/dev/cpp/valhalla/test/gurka/test_gtfs.cc: In member function ‘virtual void GtfsExample_WriteGtfs_Test::TestBody()’:
[build] /home/nilsnolde/dev/cpp/valhalla/test/gurka/test_gtfs.cc:36:3: error: designator order for field ‘gtfs::Stop::stop_name’ does not match declaration order in ‘gtfs::Stop’
[build]    36 |   };
[build]       |   ^
[build] /home/nilsnolde/dev/cpp/valhalla/test/gurka/test_gtfs.cc:44:3: error: designator order for field ‘gtfs::Stop::stop_name’ does not match declaration order in ‘gtfs::Stop’
[build]    44 |   };
[build]       |   ^
[build] /home/nilsnolde/dev/cpp/valhalla/test/gurka/test_gtfs.cc:71:5: error: designator order for field ‘gtfs::StopTime::stop_sequence’ does not match declaration order in ‘gtfs::StopTime’
[build]    71 |     };
[build]       |     ^
[build] ninja: build stopped: subcommand failed.
[build] Build finished with exit code 1
```

> it seemed like the code didn't know how to write the files until I created the directory manually. 

that's a pity, was hoping `just_gtfs` would create the dir for us when trying to write files. then we can do it as well in our code, see https://github.com/valhalla/valhalla/blob/817d60bdf69d6581b7e643737fe2c37065cbba81/valhalla/filesystem.h#L383

File: test/gurka/test_gtfs.cc [hunk 6]
```diff
@@ -0,0 +1,149 @@
+#include "gurka.h"
+#include "third_party/just_gtfs/include/just_gtfs/just_gtfs.h"
```
Comment (kevinkreiser) [inline test/gurka/test_gtfs.cc:2]: this should work if the cmake changes are correct (need to make changes in cmakelists in tests dir)
```suggestion
#include "just_gtfs/just_gtfs.h"
```

**Review by nilsnolde — DISMISSED** [review]

File: src/mjolnir/valhalla_fetch_transit.cc [hunk 7]
```diff
@@ -1181,4 +1181,4 @@ int main(int argc, char** argv) {
   stitch(pt, all_tiles, dangling_tiles);
 
   return 0;
-}
+}
```
Comment (kevinkreiser) [inline src/mjolnir/valhalla_fetch_transit.cc:1184]: you can revert this file by doing `git checkout origin/master -- src/mjolnir/valhalla_fetch_transit.cc`

**Review by kevinkreiser — CHANGES_REQUESTED** [review]
still one nitpick

**Review by nilsnolde — APPROVED** [review]

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**